pip/commands/cache.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import absolute_import
+
+import logging
+import os
+
+from pip.basecommand import Command
+from pip.status_codes import SUCCESS, ERROR
+
+
+logger = logging.getLogger(__name__)
+
+
+class CacheCommand(Command):
+    """\
+    Show information about pip's wheel cache.
+
+    Subcommands:
+        location: Print the location of the wheel cache."""
+    actions = ["location"]
+    name = "cache"
+    usage = """
+      %%prog [options] %s""" % "|".join(actions)
+    summary = "Show information about pip's wheel cache."
+
+    def run(self, options, args):
+        if not args or args[0] not in self.actions:
+            logger.warning(
+                "ERROR: Please provide one of these subcommands: %s" %
+                ", ".join(self.actions))
+            return ERROR
+        method = self.__getattribute__("action_%s" % args[0])
+        return method(options, args[1:])
+
+    def action_location(self, options, args):
+        logger.info(os.path.join(options.cache_dir, "wheels"))
```

Comment:
Yea I definitely think everything should encompass all of the caches we have. The reason I was asking what your use case was for that command is trying to determine if it makes sense to have ``pip cache location`` return the top level cache directory, or if it made sense to have flags to return specific cache directories or if it made sense to just omit the command all together and roll whatever specific functionality you were looking at using that command for into it's own command (e.g. if you were looking to do ``rm -rf $(pip cache location)`` then just add ``pip cache clear``).

However, I think the way to do it, assuming we can't just roll whatever functionality up, is maybe by default have it return the top level cache directory, and then add flags for ``--wheel`` and ``--http`` (or ``--type [wheel|http]``) that let you scope that down to a specific type instead of the entire thing. Could easily apply this pattern to other commands like ``pip cache info`` (only show info for wheel or http) and ``pip cache clear`` (only clear one type).